models/evc.py

```diff
@@ -1058,32 +1079,107 @@ def run_sdntrace(uni):
-
-        for link, trace1, trace2 in zip(self.current_path,
-                                        trace_a[1:],
-                                        trace_z[:0:-1]):
-            if compare_endpoint_trace(
-               link.endpoint_a,
-               glom(link.metadata, 's_vlan.value'), trace2) is False:
-                log.warning(f"Invalid trace from uni_a: {trace_a}")
-                return False
-            if compare_endpoint_trace(
-               link.endpoint_b,
-               glom(link.metadata, 's_vlan.value'), trace1) is False:
-                log.warning(f"Invalid trace from uni_z: {trace_z}")
-                return False
+    @staticmethod
+    def run_bulk_sdntraces(uni_list):
+        """Run SDN traces on control plane starting from EVC UNIs."""
+        endpoint = f"{settings.SDN_TRACE_CP_URL}/traces"
+        data = []
+        for uni in uni_list:
+            data_uni = {
+                "trace": {
+                            "switch": {
+                                "dpid": uni.interface.switch.dpid,
+                                "in_port": uni.interface.port_number,
+                            }
+                        }
+                }
+            if uni.user_tag:
+                data_uni["trace"]["eth"] = {
+                                            "dl_type": 0x8100,
+                                            "dl_vlan": uni.user_tag.value,
+                                            }
+            data.append(data_uni)
+        response = requests.put(endpoint, json=data, timeout=30)
+        if response.status_code >= 400:
+            log.error(f"Failed to run sdntrace-cp: {response.text}")
+            return []
+        return response.json()
 
-        return True
+    @staticmethod
+    # pylint: disable=too-many-locals
+    def check_list_traces(list_circuits):
+        """Check if current_path is deployed comparing with SDN traces."""
+        uni_list = []
+        circuit_data = {}
+        for circuit_id in list_circuits:
+            circuit = list_circuits[circuit_id]
+            uni_list.append(circuit.uni_a)
+            uni_list.append(circuit.uni_z)
+            dpid_a = circuit.uni_a.interface.switch.dpid
+            port_a = circuit.uni_a.interface.port_number
+            interface_a = str(dpid_a) + ':' + str(port_a)
+            circuit_data[interface_a] = {
+                                            'circuit_id': circuit.id,
+                                            'trace_name': 'trace_a'
+                                        }
+            dpid_z = circuit.uni_z.interface.switch.dpid
+            port_z = circuit.uni_z.interface.port_number
+            interface_z = str(dpid_z) + ':' + str(port_z)
+            circuit_data[interface_z] = {
+                                            'circuit_id': circuit.id,
+                                            'trace_name': 'trace_z'
+                                        }
+        if len(uni_list) == 0:
```

Comment:
I think you could make the early return intention clearer. The only way for `uni_list` still be empty is if `list_circuits` is empty too, so you could have this early return in the first lines of this method making the intention clearer with `if not list_circuits: return {}`. Could you adapt this?